applications/callflow/src/module/cf_resources.erl

```diff
@@ -341,3 +344,26 @@ handle_channel_destroy(<<"loopback", _/binary>>, _JObj) ->
     {<<"TRANSFER">>, 'ok'};
 handle_channel_destroy(_, JObj) ->
     {kz_call_event:hangup_cause(JObj), kz_call_event:hangup_code(JObj)}.
+
+-spec get_privacy_prefs(kapps_call:call()) -> kz_term:proplist().
+get_privacy_prefs(Call) ->
+    lager:debug("Checking inception of call"),
+    case kapps_call:inception(Call) of
+        'undefined' -> get_privacy_prefs_from_endpoint(Call);
+        _Else -> []
+    end.
+
+-spec get_privacy_prefs_from_endpoint(kapps_call:call()) -> kz_term:proplist().
+get_privacy_prefs_from_endpoint(Call) ->
+    lager:debug("Call is outbound, checking caller_id_outbound_privacy value"),
+    {'ok', Endpoint} = kz_endpoint:get(Call),
+    case kz_json:get_value(<<"caller_id_outbound_privacy">>, Endpoint) of
```

Comment:
@bpar476 Howdy!  This looks great with one exception.  We prefer a hierarchical data structure to 'group' like parameters and have an existing object called 'caller_id_options' that we believe is a more appropriate location for this.  More generally we are also avoiding adding non-object root level properties to documents for a number of reasons :)

As such, we request that you move this to caller_id_options.outbound_privacy, let us know your thoughts.  Thanks!

Reply:
Will get right onto it and comment again when I have updated the pull request :smile: